internal/verifier/migration_verifier.go

```diff
@@ -399,8 +399,10 @@ func (verifier *Verifier) getGenerationWhileLocked() (int, bool) {
 
 func (verifier *Verifier) maybeAppendGlobalFilterToPredicates(predicates []bson.D) []bson.D {
 	if verifier.globalFilter == nil {
+		verifier.logger.Debug().Str("filter", fmt.Sprintf("%v", verifier.globalFilter)).Msg("No filter to append")
 		return predicates
 	}
+	verifier.logger.Debug().Str("filter", fmt.Sprintf("%v", verifier.globalFilter)).Msg("Appending filter to find query")
```

Comment:
`globalFilter` is a single `bson.D` unmarshaled from a JSON object. Although it can contain multiple fields, I don't think the user can put a slice of filters.